torchft/manager.py

```diff
@@ -324,6 +328,21 @@ def __init__(
         # first step is 1
         self._participating_replica_rank: Optional[int] = None
         self._participating_replica_world_size: int = 0
+        self._is_state_dict_read_allowed = True
+
+    def allow_state_dict_read(self) -> None:
```

Comment:
Can we add specific tests for these methods? Lock logic is pretty risk prone so would be nice to have unit test coverage for these